pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -208,14 +210,43 @@ func (d *DeploymentController) configureIstioGateway(log *istiolog.Scope, gw gat
 	}
 	log.Info("reconciling")
 
-	svc := serviceInput{Gateway: &gw, Ports: extractServicePorts(gw)}
-	if err := d.ApplyTemplate("service.yaml", svc); err != nil {
+	defaultName := getDefaultName(gw.Name, &gw.Spec)
+	gatewayName := defaultName
+	if nameOverride, exists := gw.Annotations[gatewayNameOverride]; exists {
+		gatewayName = nameOverride
+	}
+
+	gatewaySA := defaultName
+	if saOverride, exists := gw.Annotations[gatewaySAOverride]; exists {
+		gatewaySA = saOverride
+	}
+
+	input := MergedInput{
+		Gateway:        &gw,
+		GatewayName:    gatewayName,
+		ServiceAccount: gatewaySA,
+		Ports:          extractServicePorts(gw),
+		KubeVersion122: kube.IsAtLeastVersion(d.client, 22),
+	}
+
+	ingressSa := d.RenderServiceAccountApply(input)
+	_, err := d.client.Kube().
+		CoreV1().
+		ServiceAccounts(gw.Namespace).
+		Apply(context.Background(), ingressSa, metav1.ApplyOptions{
```

Comment:
When was `Apply` introduced in k8s？ Is that version before the scope of istio supported versions

Reply:
I think it's 1.18 which is very old. But this feature already used Apply before this PR (just via a different client-go method, same API call)

Reply:
OK, got it. it is k8s serverside apply